### ISSUE TITLE
Unsatisfiable classes revealed by ELK 0.5

### ISSUE THREAD

**Issue by balhoff** [obophenotype/uberon#1455]
Classifying with ELK 0.5, which supports property ranges, reveals a pretty large number of unsatisfiable classes (5590) when the full RO is imported into Uberon (release `http://purl.obolibrary.org/obo/uberon/releases/2018-07-30/ext.owl`). You've got to import RO to see the problems.

**Comment by balhoff** [comment]
I think this one is fairly typical:

<img width="931" alt="futurecoelomiccavitylumen" src="https://user-images.githubusercontent.com/210210/45228769-17427200-b292-11e8-8528-d5b7a599b79e.png">

**Comment by balhoff** [comment]
Deleting the axiom `'material entity' DisjointWith 'immaterial entity'` eliminates all but 4 unsatisfiable classes.

**Comment by matentzn** [comment]
Yeah, I thought that might have been the cause. I found a bunch of those where anatomical lines and anatomical structures were mixed and matched together... Solving this will be very important.

**Comment by balhoff** [comment]
I guess the pivotal axiom is really `'has developmental contribution from' Range 'anatomical structure'`.

**Comment by cmungall** [comment]
Yes, I think this should be generalized. Lexically this sounds strange 
to talk about spaces 'contributing', but the intent of the relation is 
literally the property chain hp o df

**Comment by balhoff** [comment]
The range of 'has developmental contribution from' in RO has been loosened to 'anatomical entity'. This seems to have resolved nearly all of these problems. There are 23 remaining unsatisfiable classes (when you import both RO and uberon-caro-bridge into Uberon). The issue is things attached to a border of a bone, where the border is defined as an anatomical line (immaterial entity):

![deltoid](https://user-images.githubusercontent.com/210210/54944786-a9e2ef80-4f0a-11e9-9875-e41680548f1a.png)

**Comment by cmungall** [comment]
OK, I think borders should be material, will change

**Comment by balhoff** [comment]
Only one problem class right now:

![subscapularis](https://user-images.githubusercontent.com/210210/127907155-a77b47a3-865b-4de5-93dc-9df0c29df871.png)

**Comment by cmungall** [comment]
I think for now, we change the attachment site to be the scapula, and add a note that it's in the fossa

Removing incoherencies is number one priority, fine grained axiomatization can be added iteratively.

I made a ticket for the broader issues #2017.   but this may take a while to resolve

**Comment by matentzn** [comment]
I neither see this showing up in uberon.owl nor uberon-edit.obo.. Which file/reasoner are you using?

**Comment by balhoff** [comment]
You need to import the full `ro.owl` and also `uberon-bridge-to-caro`.

**Comment by matentzn** [comment]
https://github.com/obophenotype/uberon/pull/2023 indeed fixes the issue..
